ssc/cmod_battery.cpp

```diff
@@ -1231,29 +1232,25 @@ void battstor::parse_configuration()
         prediction_index = 0;
         if (batt_meter_position == dispatch_t::BEHIND)
         {
-            if (batt_dispatch == dispatch_t::LOOK_AHEAD || batt_dispatch == dispatch_t::MAINTAIN_TARGET || batt_dispatch == dispatch_t::FORECAST)
+            if (batt_dispatch == dispatch_t::PEAK_SHAVING || batt_dispatch == dispatch_t::MAINTAIN_TARGET || batt_dispatch == dispatch_t::FORECAST)
             {
-                look_ahead = true;
+                look_ahead = batt_weather_forecast == dispatch_t::WEATHER_FORECAST_CHOICE::WF_LOOK_AHEAD;
+                look_behind = batt_weather_forecast == dispatch_t::WEATHER_FORECAST_CHOICE::WF_LOOK_BEHIND;
+                input_forecast = batt_weather_forecast == dispatch_t::WEATHER_FORECAST_CHOICE::WF_CUSTOM;
                 if (batt_dispatch == dispatch_t::MAINTAIN_TARGET)
                     input_target = true;
             }
             else if (batt_dispatch == dispatch_t::CUSTOM_DISPATCH)
             {
                 input_custom_dispatch = true;
             }
-            else
-                look_behind = true;
         }
         else if (batt_meter_position == dispatch_t::FRONT)
         {
-            if (batt_dispatch == dispatch_t::FOM_LOOK_AHEAD || batt_dispatch == dispatch_t::FOM_PV_SMOOTHING) {
-                look_ahead = true;
-            }
-            else if (batt_dispatch == dispatch_t::FOM_LOOK_BEHIND) {
-                look_behind = true;
-            }
-            else if (batt_dispatch == dispatch_t::FOM_FORECAST) {
-                input_forecast = true;
+            if (batt_dispatch == dispatch_t::FOM_AUTOMATED_ECONOMIC || batt_dispatch == dispatch_t::FOM_PV_SMOOTHING) {
+                look_ahead = batt_weather_forecast == dispatch_t::WEATHER_FORECAST_CHOICE::WF_LOOK_AHEAD;
+                look_behind = batt_weather_forecast == dispatch_t::WEATHER_FORECAST_CHOICE::WF_LOOK_BEHIND;
+                input_forecast = batt_weather_forecast == dispatch_t::WEATHER_FORECAST_CHOICE::WF_CUSTOM;
```

Comment:
same, `look_ahead`, etc should all be False to start then one should be toggled on

Reply:
These are flagged to false in the header file. I'll proceed with the case statement for clarity.